test/unittests/commands/xmlupload/test_value_transformers.py

```diff
@@ -21,16 +28,59 @@
         (False, False),
     ],
 )
-def test_transform_boolean_success(input_bool: Any, success: bool) -> None:
-    result = transform_boolean(input_bool)
-    assert result == success
+def test_transform_boolean_success(input_bool: Any, out_bool: bool) -> None:
+    result = transform_xsd_boolean(input_bool)
+    assert result == Literal(out_bool, datatype=XSD.boolean)
 
 
 def test_transform_boolean_raises_string() -> None:
     with pytest.raises(BaseError):
-        transform_boolean("foo")
+        transform_xsd_boolean("foo")
 
 
 def test_transform_boolean_raises_int() -> None:
     with pytest.raises(BaseError):
-        transform_boolean(2)
+        transform_xsd_boolean(2)  # type: ignore[arg-type]
+
+
+def test_transform_xsd_decimal_int() -> None:
+    result = transform_xsd_decimal("1")
+    assert result == Literal("1.0", datatype=XSD.decimal)
+
+
+def test_transform_xsd_decimal_decimal() -> None:
+    result = transform_xsd_decimal("1.1")
+    assert result == Literal("1.1", datatype=XSD.decimal)
+
+
+def test_transform_interval_success() -> None:
+    result = transform_interval("1:2")
+    assert isinstance(result, Interval)
+    assert result.start == Literal("1.0", datatype=XSD.decimal)
+    assert result.end == Literal("2.0", datatype=XSD.decimal)
+
+
+def test_transform_interval_success_with_space() -> None:
+    result = transform_interval("1:2.1 ")
```

Comment:
I tried it out in the XML, it is not possible to enter a non integer value. The xsd file also prevents empty strings. Therefore, it is not possible to have any other inputs.

Reply:
The xsd file can be changed (willingly or by mistake) in the future, so we shouldn't rely too much on it. But for this time, I think it's okay.